services/tracer/TracingSesameWindowed.cpp

```diff
@@ -0,0 +1,68 @@
+#include "services/tracer/TracingSesameWindowed.hpp"
+
+namespace services
+{
+    TracingSesameWindowed::TracingSesameWindowed(SesameEncoded& delegate, Tracer& tracer)
+        : SesameWindowed(delegate)
+        , tracer(tracer)
+    {}
+
+    void TracingSesameWindowed::ReceivedInit(uint16_t newWindow)
+    {
+        tracer.Trace() << "SesameWindowed::ReceivedInit newWindow: " << newWindow;
```

Comment:
Don't you need to pass the calls to base class?? I don't see any base calls :P

Reply:
No, these functions are empty in the base class, and their sole purpose is to report activity so that it can be traced. This is an example of the template method pattern.